app/src/main/java/com/bilal/cactus_notes_project/ui/SignupActivity.kt

```diff
@@ -0,0 +1,95 @@
+package com.bilal.cactus_notes_project.ui
+
+import android.os.Bundle
+import android.util.Patterns
+import androidx.appcompat.app.AppCompatActivity
+import com.bilal.cactus_notes_project.R
+import com.bilal.cactus_notes_project.databinding.ActivitySignupBinding
+
+
+class SignupActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivitySignupBinding
+    private val userNameRegex = "^[a-z0-9_]*$".toRegex()
+    private val passwordRegex = "^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%^&+=]).+$".toRegex()
```

Comment:
This regex enforces order on digits, lowercase, uppercase and special characters. Hence does not work correctly.

Notice it doesn't match:

<img width="897" alt="Screen Shot 2021-07-14 at 17 28 41" src="https://user-images.githubusercontent.com/4990386/125639694-0ff7c1ee-579e-4d10-ba3f-e786187c3b01.png">

Reply:
DONE